### DISCUSSION TITLE
Whatsapp Ban

### DISCUSSION THREAD

**Discussion by deven96** [tulir/whatsmeow#199] [WhatsApp protocol Q&A]
Wrote up a bot using a number from Wabi and this library. How do I prevent being banned by Whatsapp as the traffic increases

**Comment by mamur-rezeki** [comment]
This out of the library context. Maybe by managing our "sending" traffic could be prevent getting banned by WhatsApp.

**Reply by deven96** [reply]
Understandably, which is why I opened as a discussion and not an issue so I can know if others who have used the library have experienced it or not

**Reply by kalijassy** [reply]
I am  facing WhatsApp ban problem from 3 months if u know kindly help +917347261998 whattsapp

**Comment by abbasudo** [comment]
@deven96 can you tell me if you did get banned? how many messages did you send before getting ban if so

**Reply by deven96** [reply]
I did

**Reply by abbasudo** [reply]
what are the limits?

**Reply by deven96** [reply]
Limits? 
Not certain... Just tried to open the app and saw the ban

**Comment by Stein-Hakase** [comment]
You probably sending messages to users that aren't in your contacts list

**Comment by hrizal** [comment]
Ban Reason
"**too many people blocked you**",
"**you sent too many messages to people who don't have you in their address books**"
"**you created too many groups with people who don't have you in their address books**"
"**you sent the same message to too many people**"
"**you sent too many messages to a broadcast list**"

**Reply by hajsf** [reply]
Any quantitative number of the "too many"?

**Reply by muhammadrivaldy** [reply]
Interesting, is this your opinion or based on the terms and conditions of whatsapp?
cc: @hrizal

**Comment by roniahmad** [comment]
I tried to send a message to 30 number which is not in my contact list, with delay 30 seconds each message (try to mimic human), and still get banned.

I used two number, one as a sender, and one as a receiver. I tried to send 1000 messages (without delay) to my receiver number without getting banned, even I tried multiple times.

if there is a solution or workaround of this "issue", it will be a big help

**Reply by sp3c7r4** [reply]
Have you tried sending it with random delays

**Reply by martinsjr** [reply]
It doesn't make sense to send with random delays if your identity is completely generic... have you looked at the code?

**Reply by martinsjr** [reply]
Ask yourself this question...
If you get banned very quickly, the problem lies in your initial actions... what are they?

1. Hello (Client/Server)

Are you really what you want to appear to be?

**Comment by hrizal** [comment]
Return to the terms and conditions that caused you to be banned. and it's
not a technical problem or various tricks to avoid it, as long as there is
a violation, bans are something that cannot be avoided

Pada Jum, 22 Nov 2024 10.03, Roni Ahmad ***@***.***> menulis:

**Comment by giuseongit** [comment]
From what you're saying, it seems that using whatsmeow for normal, business oriented use (which could be a simple business automation system) should not trigger the ban... very interesting.
I'll make some tests over a long period of time and then report here some results. I'm writing a notification system for appointments.

**Reply by compounder26** [reply]
any update on this @giuseongit ?

**Reply by giuseongit** [reply]
The bot is up and running, no bans so far. I'm sending messages only to contacts which are arleady in the address book of the account, in a 9-12am time window, with at least 500ms between messages, trying to mimick a real user usage.

**Comment by wxnnvs** [comment]
I made a ping pong autoresponder on my main number

I still use it daily and even tho some people have spammed it like crazy i domt have any issues (used only by my comtacts of rn)

Maybe if they text first its less of a violation

**Reply by roniahmad** [reply]
> Maybe if they text first its less of a violation

yeah I knew about this. This is not the issue when your app/service acts as a "responder"


> even tho some people have spammed it like crazy i domt have any issues

This is true. The problem occurs when you make your service as a "broadcaster", they never reach you out before and suddenly you sent them message. Even tho you tried to mimic human(put delay, add 'random signature words' or else) , you'll get banned

**Reply by wxnnvs** [reply]
Interesting,
I guess you should let clients initiate the convo if u want to build somethign without getting banned next day

**Reply by deven96** [reply]
> Interesting, I guess you should let clients initiate the convo if u want to build somethign without getting banned next day

I dont think that's the case as my project https://github.com/deven96/whatsticker, which i used this library to build as at the time, got blocked regardless. I never reached out first, as the sole purpose was to only respond with a sticker if media was sent. Years later, whatsapp actually made a sticker button so this is fine to leave on the shelf but I never quite figured out the ban
